### ISSUE TITLE
exclude option from config file is ignored when run for single file with githook

### ISSUE THREAD

**Issue by NKorchagin** [nicklockwood/SwiftFormat#374]
That's the issue we faced recently. 

We have configuration file with exclude option:
```
--exclude **/Generated
```
If you run swiftformat for root folder like that:
```
swiftformat . --config swiftformat.config
```
files in Generated folder will be ignored correctly.

But, if you run it for specific file, like we do in githook for all staged files:
```
swiftformat --config swiftformat.config "SomeFolder/Generated/SomeFile.swift"
```
This file will be formatted.

After investigation we found that issue was introduced in version `0.36.0`. Version `0.35.10` will say `error: no eligible files found at .../SomeFolder/Generated/SomeFile.swift` but will not format this file.

@nicklockwood please, review that change, probably it's the reason
https://github.com/nicklockwood/SwiftFormat/commit/10cafc4af83b596dbf7ee932d421d68b15006e5e

**Comment by nicklockwood** [comment]
@NKorchagin thanks for reporting this - I'll get it fixed asap.

**Comment by nicklockwood** [comment]
@NKorchagin fixed in 0.39.4

**Comment by palle-k** [comment]
Possible regression in 0.46.3: When I run `swiftformat --config .swiftformat ./**/*.swift` instead of `swiftformat --config .swiftformat ./`, my exclude list will be ignored.

**Comment by nicklockwood** [comment]
@palle-k using wildcards for input paths is not currently supported. I guess this works because the shell itself expands the path, but it's not something I ever tested. Are you sure it worked prior to 0.46.3?

**Comment by palle-k** [comment]
I haven't tested it with prior versions. But I have noticed that this error also occurs in the pre-commit hook and without wildcards.

**Comment by nicklockwood** [comment]
@palle-k ah OK, thanks. I'll take a look.

**Comment by palle-k** [comment]
Thanks. I was able to work around it for now by using exclude patterns directly with `git-format-staged` (e.g. `"!Pods/*"`) but a solution directly in SwiftFormat would be ideal.

**Comment by nicklockwood** [comment]
@palle-k with `git-format-staged` are you using the `--stdinpath` argument? I recently updated the docs to recommend this, as it is required for exclusions to work correctly:

```
git-format-staged --formatter "swiftformat stdin --stdinpath '{}'" "*.swift"
```

(the reason for this is that `git-format-staged` passes files via stdin instead of passing the file path itself, so without the `--stdinpath` argument, swiftformat cannot know what the path of the input was, and whether to exclude it)

**Comment by palle-k** [comment]
Yes I am.

**Comment by nicklockwood** [comment]
@nicklockwood can you try it without the `--config path/to/.swiftformat`? It should pick that up anyway if it's in the same directory.

**Comment by palle-k** [comment]
The Pods directory is getting formatted even without the `--config` argument.

**Comment by oonoo** [comment]
This is not working with the current version 0.48.1. Is it possible to open this issue again? Or another way to make it visible this could need a fix?

**Comment by keithcml** [comment]
@oonoo which version is workable do you know?

@nicklockwood any updates for this issues?

**Comment by nicklockwood** [comment]
@keithcml sorry, I forgot about this. AFAICT there's no way to bump reopened issues to the front page of the github issues, so I've [opened a new issue](https://github.com/nicklockwood/SwiftFormat/issues/938) for this. Please post any further comments there.

**Comment by nicklockwood** [comment]
@keithcml @oonoo @palle-k fixed in 0.48.3, sorry for the delay
